composables/useErc725.ts

```diff
@@ -39,17 +44,211 @@ const fetchProfile = async (profileAddress: Address): Promise<Profile> => {
     ? getImageUrlBySize(lsp3Profile.backgroundImage, 800, 400)
     : ''
 
+  const { getBalance } = useWeb3(PROVIDERS.RPC)
+  const balance = await getBalance(profileAddress)
+
   return {
     ...lsp3Profile,
     address: profileAddress,
     profileImageUrl: optimalProfileImage || '',
     backgroundImageUrl: optimalBackgroundImage || '',
+    balance,
+  }
+}
+
+const fetchAssets = async (profileAddress: Address, schema: string) => {
+  const erc725 = getInstance(
+    profileAddress,
+    LSP3UniversalProfileMetadata as ERC725JSONSchema[]
+  )
+  const result = await erc725.fetchData(schema)
+  const assetAddresses = result.value as Address[]
+  const { profile } = useProfileStore()
+  console.log(schema, assetAddresses)
+
+  const assets = Promise.all(
+    assetAddresses.map(async address => {
+      const standard = await detectStandard(address)
+      let data
+
+      switch (standard) {
+        case 'LSP8IdentifiableDigitalAsset': {
+          assertAddress(profile.address)
+          data = await fetchLSP8Assets(address, profile.address)
```

Comment:
can this throw? If so, I think we will lose all the promise.

Maybe promise.allsettled could be a bit more solid?